### DISCUSSION TITLE
Difference between `scitype="time-series-regressor"` and `"tabular-regressor"` (`TSFeatureExtractor`)

### DISCUSSION THREAD

**Discussion by ngupta23** [alan-turing-institute/sktime#3841] [Q&A]
I am trying to follow the example of the `TSFeatureExtractor` as used for forecasting and have a question. If I use `scitype="time-series-regressor"`, then the reduced regression works with `TSFeatureExtractor`. But if I change the `scitype="tabular-regressor"`, then I get an error. I would like to understand what the difference is between these two scitypes and when to use which one. Thanks for any information you can provide.

```python
from sklearn.ensemble import RandomForestRegressor
from sktime.datasets import load_airline
from sktime.forecasting.base import ForecastingHorizon
from sktime.forecasting.compose import make_reduction
from sktime.forecasting.model_selection import temporal_train_test_split
from sktime.transformations.series.summarize import WindowSummarizer
from sktime.transformations.panel.tsfresh import (
    TSFreshFeatureExtractor,
    TSFreshRelevantFeatureExtractor,
)
from sklearn.pipeline import make_pipeline
y = load_airline()
y_train, y_test = temporal_train_test_split(y)
fh = ForecastingHorizon(y_test.index, is_relative=False)
regressor = make_pipeline(
    TSFreshFeatureExtractor(show_warnings=False, disable_progressbar=True),
    RandomForestRegressor(),
)

# From sktime documentation (this works) ----
forecaster = make_reduction(
    regressor, scitype="time-series-regressor", window_length=12
)
forecaster.fit(y_train)
y_pred = forecaster.predict(fh)

# Matched to pycaret reduced regressor (this does not work) ----
forecaster2 = make_reduction(
    regressor, scitype="tabular-regressor", strategy="recursive", window_length=12
)
forecaster2.fit(y_train)
y_pred2 = forecaster2.predict(fh)
```

**Comment by fkiraly** [comment]
@ngupta23, the `scitype` argument is a bit of a legacy thing and users should avoid using it.

The "best case end state" for `make_reduction` is that it recognizes what (sci)type of regressor (time series or tabular) the `regressor` argument is, and `make_reduction` applies the right reduction wrapper under the hood (tabular or time series).

Historically, very far in the past, `make_reduction` did not implement a type check, so users had to specify the dispatch path by the `scitype` argument.

Right now on `main`, I thought it was redundant (because the base classes etc allow this inference in-principle), but in the implementation it turns out it was not, which is probably the reason the implementers still left it there. (if you are interested in the detail: `Pipeline`-s in `sklearn` are polymorphic, so the inference is non-trivial - currently it fails for the special case of `sklearn` `Pipeline`-s where it has to be specified explicitly)

Still, inference is possible, it was just not implemented by whoever had worked on it.

So, I've just implemented it in various PR and removed the `scitype` argument from remaining tests and examples:

* https://github.com/sktime/sktime/pull/4020
* https://github.com/sktime/sktime/pull/4021
* https://github.com/sktime/sktime/pull/4022

From now on, users should not need to specify it (because it is always correctly inferred).

Thanks for bringing this up and to our notice!
